src/System.Device.Gpio/Interop/Unix/SafeChipHandle.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio
+{
+    internal class SafeChipHandle : SafeHandle
+    {
+        public SafeChipHandle() : base(IntPtr.Zero, true) { }
+
+        protected override bool ReleaseHandle()
+        {
+            Interop.CloseChip(handle);
+            return true;
+        }
+
+        public override bool IsInvalid => handle == IntPtr.Zero || handle == new IntPtr(-1);
```

Comment:
Do we always have to new up an IntPtr for every call to IsInvalid or can we cache it in a static?

Reply:
Also, when will handle ever `== IntPtr(-1)`?

Reply:
> Do we always have to new up an IntPtr for every call to IsInvalid or can we cache it in a static?

Moved to static variable

Not sure what about the 2nd question, if you asking when handle could be equal to IntPtr(-1), then I guess when calling native method for chip pointer it might return -1 in some error nous conditions.